Libraries/Image/RCTImageLoader.m

```diff
@@ -74,8 +74,7 @@ + (void)loadImageWithTag:(NSString *)imageTag callback:(void (^)(NSError *error,
           // doesn't spike the memory up during the process.
           @autoreleasepool {
             ALAssetRepresentation *representation = [asset defaultRepresentation];
-            ALAssetOrientation orientation = [representation orientation];
```

Comment:
It seems okay to ignore the asset orientation in this case because `-[ALAsset thumbnail]` should return a correctly oriented image. I still don't think this is a the correct solution though.